components/lsp-form/helpers.ts

```diff
@@ -11,3 +11,25 @@ export const calculateTimeRemaining = () => {
 
   return text;
 };
+
+// Only permits numbers and decimals
+// This is intended for a keydown or keyup event handler.
+// Tab, Backspace, and copy+paste also goes through.
+export const onlyAllowNumbersAndDecimals = (
+  event: React.KeyboardEvent<HTMLInputElement>
```

Comment:
not totally convinced this is the way to do this, vs checking the entire string as a valid number. you can have situations now with multiple `.` like `...........1`.  to prevent this its easiest to check the whole input

Reply:
I think both is probably the optimal strategy.

We can definitely show error messages for bad values, but if you are trying to restrict values it's very difficult to say.. do a replace *as* they type, unless it's a very simple replace.

For instance, when they type "." is this a good or bad value? well it totally depends on how many "." there are. And because this also effects the two other inputs, it gets a bit awkward what you do.

But we can easily enforce value validation for characters they can legally put in. (It should be noted too, I am allowing them to copy paste with that function so they could bypass the character limitation.. but I think this is ok. If you think a bunch of text will post properly to the blockchain it's your ETH to spend on gas).